Add Atari version of of doesclrscrafterexit().

groessler opened the pull request:
- Update documentation.
- Update atari.h and apple2.h header files.
- Adapt Atari test/target programs.
- Fix a typo in "div" entry in funcref.sgml.

greg-king5 commented:
"numer", "denom", "quot", and "rem" are abbreviations of "numerator", "denominator", "quotient", and "remainder" -- names used by mathematicians when describing integer division.  It isn't a typo in "funcref.sgml".

greg-king5 commented:
Oliver suggested that `doesclrscrafterexit()` be declared in `<cc65.h>`.  And, a default version of it should be put in `libsrc/common/`; it always will return false (`_doesclrscrafterexit := return0`).  `libsrc/Makefile` will replace that one with a platform-specific version if it exists.  That will let us use the test function in our programs without needing to use `#if defined(<__platform__>)`.

oliverschmidt commented on include/apple2.h:
```diff
@@ -177,6 +177,9 @@ unsigned char get_ostype (void);
 /* Reboot machine after program termination has completed. */
 
+unsigned char doesclrscrafterexit (void);
+/* Will the screen automatically be cleared after program termination. */
```
Maybe a question mark instead of the period.

oliverschmidt commented on doc/funcref.sgml:
```diff
@@ -2587,6 +2567,40 @@ ldiv
+<quote>
+<descrip>
+<tag/Function/Determines whether the screen is going to be cleared after program exit.
+<tag/Header/<tt/<ref id="atari.h" name="atari.h">, <ref id="apple2.h" name="apple2.h">/
```
This needs to reference cc65.h - see Gregs comment.

oliverschmidt commented on doc/funcref.sgml:
```diff
@@ -2587,6 +2567,40 @@ ldiv
+<tag/Header/<tt/<ref id="atari.h" name="atari.h">, <ref id="apple2.h" name="apple2.h">/
+<tag/Declaration/<tt/unsigned char doesclrscrafterexit (void);/
+<tag/Description/The function returns 0 if the screen won't be cleared immediately after
+program termination. It returns 1 if it will.
```
The Apple II variant doesn't return 1 but some non-0 value. The usual way to phrase that in cc65 is something like "The function returns a value of zero if <...>. It returns non zero otherwise."

oliverschmidt commented on doc/funcref.sgml:
```diff
@@ -2587,6 +2567,40 @@ ldiv
+to decide if a delay or wait for a key press should be executed when then program
+exits.
+</itemize>
+<tag/Availability/cc65 (<tt/atari/, <tt/atarixl/, <tt/apple2/, and <tt/apple2enh/ platforms)
```
As discussed above...

oliverschmidt commented on include/atari.h:
```diff
@@ -161,12 +161,13 @@ extern void __fastcall__ _scroll (signed char numlines);
+extern void _rest_vecs(void);                     /* restore system vectors */
+extern char *_getdefdev(void);                    /* get default floppy device */
+extern unsigned char _is_cmdline_dos(void);       /* does DOS support command lines */
+extern unsigned char doesclrscrafterexit (void);  /* will DOS clear the screen after program termination */
```
As discussed...

oliverschmidt commented on libsrc/atari/targetutil/w2cas.c:
```diff
@@ -35,7 +35,7 @@ static struct __iocb *findfreeiocb(void)
-    if (! _is_cmdline_dos())
+    if (doesclrscrafterexit())
         cgetc();
 }
```
Given that the return value of doesclrscrafterexit() doesn't change during runtime of the program it may be even "cooler" to just do

```
if (doesclrscrafterexit())
    atexit(cgetc);
```

groessler commented:
I've addressed the issues. I found a typo in the log message after pushing, so I did a "commit --amend" followed by a forced push. Don't know if github likes that, I was a bit hasty...

oliverschmidt commented on libsrc/common/doesclrscr.s:
```diff
@@ -0,0 +1,14 @@
+
+        .export  _doesclrscrafterexit
+
+_doesclrscrafterexit:
```
Please create an alias instead of spending bytes:

` _doesclrscrafterexit := return0 `

oliverschmidt commented on include/atari.h:
```diff
@@ -161,12 +161,12 @@ extern void __fastcall__ _scroll (signed char numlines);
+extern void _save_vecs(void);                     /* save system vectors */
+extern void _rest_vecs(void);                     /* restore system vectors */
+extern char *_getdefdev(void);                    /* get default floppy device */
+extern unsigned char _is_cmdline_dos(void);       /* does DOS support command lines */
```
Please fully revert your changes in this file.

groessler commented:
done

oliverschmidt commented:
Thanks :-)
